discovery-infra/test_infra/assisted_service_api.py

```diff
@@ -145,11 +145,21 @@ def get_hosts_in_error_status(self, cluster_id: str):
     def clusters_list(self) -> List[Dict[str, Any]]:
         return self.client.list_clusters()
 
+    def infra_envs_list(self) -> List[models.InfraEnv]:
+        return self.client.list_infra_envs()
+
     def get_all_clusters(self) -> List[Dict[str, Any]]:
         return self.client.list_clusters(get_unregistered_clusters=True)
 
     def cluster_get(self, cluster_id: str) -> models.cluster.Cluster:
         return self.client.get_cluster(cluster_id=cluster_id)
+    
+    def get_infra_env_by_cluster_id(self, cluster_id: str) -> models.infra_env.InfraEnv:
```

Comment:
There can be more then single `InfraEnv` for single cluster

Reply:
changed to return filtered list